mat64/dense.go

```diff
@@ -417,3 +420,25 @@ func (m *Dense) Augment(a, b Matrix) {
 	w := m.View(0, ac, br, bc).(*Dense)
 	w.Copy(b)
 }
+
+// MarshalBinary encodes the receiver into a binary form and returns the result.
+func (m Dense) MarshalBinary() ([]byte, error) {
+	buf := new(bytes.Buffer)
+	enc := binary.NewEncoder(buf)
+	err := enc.Encode(&m.mat)
```

Comment:
blas64 is a thin wrapper around blas. I don't think it is the job of that package to handle this. There are no methods on any of the types in that package.